### PR TITLE
Blazor Sample App

### PR DESCRIPTION
**Pull request by kilngod**
Blazor sample application

### CODE REVIEW

**Comment by m4rs-mt** [comment]
@kilngod Thanks for this great sample 🥇. Would it be possible to disable secure connections via `https` on the `localhost` side in the Blazer website configuration? This seems to cause some problems from time to time on various test machines.

**Comment by kilngod** [comment]
@m4rs-mt I'll take a look at removing the https:// from the sample app Friday.

**Comment by kilngod** [comment]
@m4rs-mt Marcel, turns out I had time this morning to sort out removing https:// actually simple to remove.

File: Samples/BlazorSampleApp/ILGPUWebHost/IComputeHost.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        /// <summary>
+        /// number of active compute sessions
+        /// </summary>
+        int SessionCount { get; }
+
+        /// <summary>
+        /// determine the minium specifications for an accelerator 
+        /// </summary>
+        /// <param name="allowedAcceleratorTypes"></param>
+        /// <param name="miniumMemory"></param>
```
Comment (m4rs-mt) [inline Samples/BlazorSampleApp/ILGPUWebHost/IComputeHost.cs:43]: minor: typo `minium`

File: Samples/BlazorSampleApp/ILGPUWebHost/ComputeHostExtension.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        /// Rank is approximated by clock rate * threads per processor * multiprocessors, for an integrated GPU 
+        /// we assume the clock rate is 2GHz to avoid operating specific calls
+        /// </summary>
+        /// <param name="accelerator"></param>
+        /// <returns></returns>
+        public static long AcceleratorRank(this Accelerator accelerator)
+        {
+            long clockRate = Convert.ToInt64(int.MaxValue); // assume 2 GHz for a CPU.
+
+            clockRate = (accelerator as ILGPU.Runtime.OpenCL.CLAccelerator)?.ClockRate ?? clockRate;
```
Comment (m4rs-mt) [inline Samples/BlazorSampleApp/ILGPUWebHost/ComputeHostExtension.cs:39]: question: are we sure that these values actually live in the same interval?

File: Samples/BlazorSampleApp/ILGPUWebHost/ComputeSession.cs [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        // one or more kernels can be hosted in each compute session depending on the complexity of our calculation
+        private System.Action<AcceleratorStream, Index1D, ArrayView1D<int, Stride1D.Dense>, ArrayView1D<float, Stride1D.Dense>, int, ArrayView<int>> mandelbrot_kernel;
+
+
+        // SessionID allows individual Blazor sessions to "find" their compute session 
+        
+        public string SessionID { get; set; }
+
+        private bool _active = true;
+        public bool IsActive { get { return _active; } set { _active = value; } }
```
Comment (m4rs-mt) [inline Samples/BlazorSampleApp/ILGPUWebHost/ComputeSession.cs:45]: nit: maybe we could simply this code to improve readability?

**Review by m4rs-mt — COMMENTED** [review]
Great work! I had some problems running this example on a computer without OpenCL or Cuda compatible devices, as the backend throws some `OutOfRange` exceptions when clicking some buttons. Should we add a note that at least an OpenCL or Cuda compatible accelerator should be available?

Another more high-level question: can we remove the pages `Basic Canvas Test` and `JavaScript notes` from the sample?

File: Samples/BlazorSampleApp/ILGPUWebHost/ComputeHostExtension.cs (hunk 2 above)
Comment (kilngod) [reply Samples/BlazorSampleApp/ILGPUWebHost/ComputeHostExtension.cs:39]: Good question... perhaps a feature to add? This is a best guess.

File: Samples/BlazorSampleApp/ILGPUWebHost/IComputeHost.cs (hunk 1 above)
Comment (kilngod) [reply Samples/BlazorSampleApp/ILGPUWebHost/IComputeHost.cs]: fixed

File: Samples/BlazorSampleApp/ILGPUWebHost/ComputeSession.cs (hunk 3 above)
Comment (kilngod) [reply Samples/BlazorSampleApp/ILGPUWebHost/ComputeSession.cs]: done

File: Samples/BlazorSampleApp/ILGPUWebHost/ComputeHostExtension.cs (hunk 2 above)
Comment (m4rs-mt) [reply Samples/BlazorSampleApp/ILGPUWebHost/ComputeHostExtension.cs:39]: Let's leave it as it is for now 👍

**Review by m4rs-mt — APPROVED** [review]
Great job 👍 Thank you for addressing my comments.

**Pull request merged**